### ISSUE TITLE
Debug in docker container connection refused

### ISSUE THREAD

**Issue by silviu-negoita** [quarkusio/quarkus#13803]
This seems more or less a regression for [this issue](https://github.com/quarkusio/quarkus/issues/7256) previously fixed.

Starting with _Quarkus 1.10_ version, following jvm arg is added when debug is enabled(dev profile activated):
`-Xrunjdwp:transport=dt_socket,address=localhost:5005,server=y,suspend=n`
P.S. Notice **localhost:5005**

This is wrong, because it will **not allow attach debug** from outside of docker container.

With _Quarkus <= 1.9.2_ (which I previously used), this jvm arg is added correctly:
`-Xrunjdwp:transport=dt_socket,address=0.0.0.0:5005`

My suggestion is either revert `address` argument, to `0.0.0.0:5005` or `*:5005`

Also, there is quick fix that can be added. On `quarkus-maven-plugin -> executions`: 
```xml
<execution>
   <phase>none</phase>
   <id>run</id>
   <goals>
      <goal>dev</goal>
   </goals>
   <configuration>
      <jvmArgs>-Xdebug -Xrunjdwp:transport=dt_socket,address=*:5005,server=y,suspend=n</jvmArgs>
      <debug>false</debug>
   </configuration>
</execution>
```

Basically debug is disabled in `quarkus-maven-plugin`, and manually added back the `jvm args` which enables it.

**Later Edit:**
Actually [this fix](https://github.com/quarkusio/quarkus/pull/13223) causes this bug.

**Comment by stuartwdouglas** [comment]
You can add <debugHost>0.0.0.0</debugHost> to the main plugin config. We won't revert to the previous behaviour as it was raised as a security/hardening issue (i.e. if you are developing on a public network someone could attach a debugger and potentially use this to compromise your machine)

**Comment by CarlosPanarello** [comment]
hi @silviu-negoita thanks for the fix, but i tried without success. How you did it to work?

I added this execution in plugin, changing the port to 8000, like the config below:
But is going up listening in 5005 when running in dev mode.

```
     <plugin>
        <groupId>io.quarkus</groupId>
        <artifactId>quarkus-maven-plugin</artifactId>
        <version>${quarkus-plugin.version}</version>
        <extensions>true</extensions>
        <executions>
          <execution>
            <goals>
              <goal>build</goal>
              <goal>generate-code</goal>
              <goal>generate-code-tests</goal>
            </goals>
          </execution>
          <execution>
            <phase>none</phase>
            <id>run</id>
            <goals>
              <goal>dev</goal>
            </goals>
            <configuration>
              <jvmArgs>-Xdebug -Xrunjdwp:transport=dt_socket,address=*:8000,server=y,suspend=n</jvmArgs>
              <debug>false</debug>
            </configuration>
          </execution>
        </executions>
      </plugin>
```

**Comment by stuartwdouglas** [comment]
Just launch with -DdebugHost=0.0.0.0 or add <configuration><debugHost>0.0.0.0</debugHost></configuration> to the plugin config.
